### PR TITLE
Compile yang configurations to binary streams

### PR DESCRIPTION
**Pull request by wingo**
This branch implements configuration compilation and loading.  It is currently a work in progress; missing items:

- [x] Support for singly-string-keyed tables

- [x] Support for raw keys but non-uniform values

- [x] Support for raw arrays

- [x] Documentation

- [x] Ability to load a configuration from a file and automatically notice if there is a corresponding up-to-date binary

- [x] Ability to automatically compile and residualize compiled configurations to disk, for later opportunistic loading

The ctable compilation is working though, which makes me very excited!!!!!  (A re-open of #531 to avoid showing older already-merged commits.)

### CODE REVIEW

**Comment by wingo** [comment]
@xray7224 this branch is ready, ptal :-))

File: src/lib/cltable.lua [hunk 1]
```diff
@@ -0,0 +1,68 @@
+module(..., package.seeall)
+
+local ffi = require("ffi")
+local ctable = require("lib.ctable")
+
+CLTable = {}
```
Comment (tsyesika) [inline src/lib/cltable.lua]: This isn't necessary I don't think.

**Review by tsyesika — COMMENTED** [review]
LGTM other than the one thing I have already brought up.

**Comment by wingo** [comment]
Thanks for the review; fixed the nit and also updated upstream SnabbCo#1059.

**Pull request merged**